time_stamper_ros/src/Node.cpp

```diff
@@ -106,3 +110,16 @@ bool Node::setGpioMode() {
   }
   return true;
 }
+
+void Node::configCallback(time_stamper_ros::LedConfig &config) {
+
+  //Ignore if not initialized to avoid memory corruption
+  if (!is_initialized_) {
+    return;
+  }
+  mode_ = static_cast<LedMode>(config.mode); //Convert int to enum
+
```

Comment:
Around here you could mutate the `config` object and set the frequency to a closest valid value.